tracing/src/main/java/com/palantir/remoting3/tracing/Tracers.java

```diff
@@ -158,6 +158,27 @@ public static Runnable wrapWithNewTrace(Runnable delegate) {
         };
     }
 
+    /**
+     * Wraps the given {@link Runnable} such that it creates a fresh {@link Trace tracing state with the given traceId}
+     * for its execution. That is, the trace during its {@link Runnable#run() execution} will use the traceId provided
+     * instead of any trace already set on the thread used to execute the runnable. Each execution of the runnable
+     * will use a new {@link Trace tracing state} with the same given traceId.
+     */
+    public static Runnable wrapWithTrace(Runnable delegate, String traceId) {
```

Comment:
do you think the usages of this would be more readable if we had the `traceId` be the first param?

```
Runnable wrappedRunnable = Tracers.wrapWithTrace(() -> {
    ...
}, traceIdToUse);
```
vs
```
Runnable wrappedRunnable = Tracers.wrapWithTrace(traceId, () -> {
    ...
});
```